### PR TITLE
Make MortarSize streamable

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Does what the title says :)

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/NumericalAlgorithms/Spectral/Projection.hpp [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace Spectral {
 /// The portion of an element covered by a mortar.
 enum class MortarSize { Full, UpperHalf, LowerHalf };
 
+std::ostream& operator<<(std::ostream& os,
+                         const MortarSize& mortar_size) noexcept;
```
Comment (nilsdeppe) [inline src/NumericalAlgorithms/Spectral/Projection.hpp]: why not take by value? enum`s are usually smaller than the pointer that the reference creates

Comment (nilsvu) [reply src/NumericalAlgorithms/Spectral/Projection.hpp]: I'll squash this ok

Comment (nilsdeppe) [reply src/NumericalAlgorithms/Spectral/Projection.hpp]: of course!

**Comment by nilsvu** [comment]
Squashed and rebased. Also fixed two more linking issues that break current develop on my macOS machine

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe did you want to look at this before it is merged?

**Review by wthrowe — APPROVED** [review]

**Pull request merged**